tests/jballerina-unit-test/src/test/resources/test-src/statements/matchstmt/const-pattern.bal

```diff
@@ -425,6 +425,38 @@ public function testConstPattern13() {
     }
 }
 
+function constPattern14(any a, any b) returns string {
+    match a {
+        "foo" => {
+            match b {
+                "bar" => {
+                    return "s1";
+                }
+                _ => {
+                    return "s2";
+                }
+            }
+        }
+        _ => {
+            match b {
+                "bar" => {
+                    return "s1";
+                }
+                _ => {
+                    return "s2";
+                }
+            }
+        }
+    }
+}
+
+function testConstPattern14() {
+    assertEquals("s2", constPattern14("foo", 2));
+    assertEquals("s1", constPattern14("foo", "bar"));
+    assertEquals("s4", constPattern14(1, 2));
+    assertEquals("s3", constPattern14(1, "bar"));
```

Comment:
Where do we get "s3" and "s4" from? Do these tests run?

I seem to be getting a compilation error for `constPattern14` - `this function must return a result` - which again is not correct because the function does return for all scenarios within the match statement.

Reply:
Fixed. I tested this with PR https://github.com/ballerina-platform/ballerina-lang/pull/27292. Therefore those changes were moved here.